ethereum/referencetests/src/main/java/org/hyperledger/besu/ethereum/referencetests/BonsaiReferenceTestWorldState.java

```diff
@@ -101,6 +206,12 @@ protected void verifyWorldStateRoot(final Hash calculatedStateRoot, final BlockH
     // The test harness validates the root hash, no need to validate in-line for reference test
   }
 
+  @Override
+  public void persist(final BlockHeader blockHeader) {
+    validateStateRolling(blockHeader);
```

Comment:
IDK how I feel about having this behavior _always_, especially in light of the evm performance metrics that are being derived from evmtool runs.  I don't know whether evmtool tests typically call persist, but if they do it would certainly be additional overhead to do state rolling.  @shemnon ?

Reply:
I thought it was just used for reference tests given the name. I will make it configurable and activate it only for the reference tests

Reply:
I will add a specific method for that and not use persist. It will avoid problems like that

Reply:
another option would be to subclass the reference test worldstate for evmtool and override the state rolling.  We could fix the class name expectation that way too

Reply:
EVMTool actually uses in-memory Forest, perhaps it should be tweaked? - https://github.com/hyperledger/besu/blob/ccab8571e5648adc00a7cec4d9a9d28155508bef/ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/BlockchainModule.java#L67

Guido's fuzzing uses in-memory Bonsai

Reply:
Yes maybe we need to update the evmtool to use bonsai or forest

Reply:
Hi @garyschulte @shemnon I updated the code to have another method for this kind of check. I think it's better to keep persist method persisting and having something else for the trielog check . Let me know what do you think about the last version

Reply:
for context https://github.com/hyperledger/besu/pull/6415/files#diff-b9e0afe79b33e1367ce04f1720fcccf35976bf2fbe93af84eb67bab40d8a0e9aR195

Reply:
> EVMTool actually uses in-memory Forest, perhaps it should be tweaked? -
> 
> https://github.com/hyperledger/besu/blob/ccab8571e5648adc00a7cec4d9a9d28155508bef/ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/BlockchainModule.java#L66-L68

I must be confused regarting evmtool vs T8nExecutor ...
https://github.com/hyperledger/besu/blob/main/ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/T8nExecutor.java#L241-L242